src/gerbt.cc

```diff
@@ -0,0 +1,313 @@
+            else {
+                ++inner;
+            }
+        }
+    }
+}
+
+// Helper functions to iterate over the butterfly indices
+void gerbt_iterate_2d(int64_t d, int64_t inner_len, int64_t mt, int64_t nt,
+                      std::function<void(int64_t, int64_t, int64_t,
+                                      int64_t, int64_t, int64_t)> body)
+{
+    // 2-sided butterflies are applied smallest to largest
+    for (int64_t k = d-1; k >= 0; --k) {
+        const int64_t num_bt = 1 << k;
+        // bt_len = ceil(nt * 2^-d) * 2^d * 2^-k
+        // half_len = 1/2 * bt_len
+        const int64_t half_len = (1 << (d-k-1))*inner_len;
+
+        for (int64_t bi = 0; bi < num_bt; ++bi) {
+            const int64_t i1 = bi*2*half_len;
+            const int64_t i2 = std::min(i1+half_len, mt);
+            const int64_t i3 = std::min(i2+half_len, mt);
+            for (int64_t bj = 0; bj < num_bt; ++bj) {
+                const int64_t j1 = bj*2*half_len;
+                const int64_t j2 = std::min(j1+half_len, nt);
+                const int64_t j3 = std::min(j2+half_len, nt);
+
+                body(i1, i2, i3, j1, j2, j3);
+            }
+        }
+    }
+}
+void gerbt_iterate_1d(Op trans, int64_t d, int64_t inner_len, int64_t mt,
+                      std::function<void(int64_t, int64_t, int64_t)> body)
+{
+    for (int64_t k_iter = 0; k_iter < d; ++k_iter) {
+        // Regular butterflies are applied largest to smallest
+        // Transposed butterflies are applied smallest to largest
+        const int64_t k = (trans == Op::NoTrans) ? k_iter : d-k_iter-1;
+
+        const int64_t num_bt = 1 << k;
+        // bt_len = ceil(mt * 2^-d) * 2^d * 2^-k
+        // half_len = 1/2 * bt_len
+        const int64_t half_len = (1 << (d-k-1))*inner_len;
+
+        for (int64_t bi = 0; bi < num_bt; ++bi) {
+            const int64_t i1 = bi*2*half_len;
+            const int64_t i2 = std::min(i1+half_len, mt);
+            const int64_t i3 = std::min(i2+half_len, mt);
+
+            body(i1, i2, i3);
+        }
+    }
+}
+
+} // namespace internal
+
+template<typename scalar_t>
+void gerbt(Matrix<scalar_t>& U_in,
+           Matrix<scalar_t>& A,
+           Matrix<scalar_t>& V)
+{
+    using BcastListTag = typename Matrix<scalar_t>::BcastListTag;
+
+    slate_assert(U_in.op() == Op::Trans);
+    slate_assert(V.op() == Op::NoTrans);
+
+    Matrix<scalar_t> U = transpose(U_in);
+
+    slate_assert(A.op() == Op::NoTrans);
+    slate_assert(U.op() == Op::NoTrans);
+    slate_assert(A.layout() == Layout::ColMajor);
+    slate_assert(U.layout() == Layout::ColMajor);
+    slate_assert(V.layout() == Layout::ColMajor);
+
+    slate_assert(U.n() == V.n());
+
+    const int64_t d = U.n();
+    const int64_t mt = A.mt();
+    const int64_t nt = A.nt();
+
+    if (d == 0) {
+        return;
+    }
+
+    int64_t inner_len = int64_t(std::ceil(nt / double(1 << d)));
+
+    #pragma omp parallel
+    #pragma omp master
+    {
+        omp_set_nested(1);
```

Comment:
I think this `omp_set_nested` is deprecated [See here](https://www.openmp.org/spec-html/5.0/openmpsu119.html)